psi4/src/psi4/libmints/basisset.cc

```diff
@@ -838,9 +784,9 @@ BasisSet::BasisSet(const std::string &basistype, SharedMolecule mol,
         center_to_nshell_[n] = nshells;
         center_to_shell_[n] = shell_count;
         Vector3 xyz = molecule_->xyz(n);
-        xyz_ptr[0] = xyz[0];
-        xyz_ptr[1] = xyz[1];
-        xyz_ptr[2] = xyz[2];
+        xyz_[3 * n + 0] = xyz[0];
```

Comment:
I'm not sure how these two are equivalent. The bit that's puzzling me is the `xyz_ptr += 3` lines below, that were made obsolete by this change - it's clear to me what the code does now. How did it work before?

Reply:
This comment and the one above are both explained by pointer syntax.
`pointer[n]` means "grab the value `n` memory locations away from where the pointer currently points" and `pointer ++ n` means "make the pointer point `n` memory locations forward". So what the old code was doing was moving the pointer ahead three every time it changed atoms, so _relative to the starting position_ it would always point at location `3 * n`.

In short - it's exactly as you said above.